code/go/0chain.net/blobbercore/handler/file_command_upload.go

```diff
@@ -198,7 +200,8 @@ func (cmd *UploadFileCommand) reloadChange(connectionObj *allocation.AllocationC
 // UpdateChange replace AddFileChange in db
 func (cmd *UploadFileCommand) UpdateChange(ctx context.Context, connectionObj *allocation.AllocationChangeCollector) error {
 	for _, c := range connectionObj.Changes {
-		if c.Operation != constants.FileOperationInsert {
+		filePath, _ := c.GetOrParseAffectedFilePath()
+		if c.Operation != constants.FileOperationInsert || cmd.fileChanger.Path != filePath {
 			continue
 		}
 		c.Size = connectionObj.Size
```

Comment:
Does this still hold?

Reply:
Yes with `cmd.fileChanger.Path != filePath` check.